# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: fbdc7fd9cb96
# Full Hash: fbdc7fd9cb96f26850f979b6f23b13f7bbaf727c
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-07-25 09:40:10
# Regressor Bug: 1652144
# File Overlap Count: 1
# Description:
#   Bug 1652144 - Part 2: Consider current BrowsingContextGroup for process selection, r=farre
#   
#   This requires keeping track of the current process used to host documents with a
#   particular remote type loaded in each BrowsingContextGroup. Due to lifecycle
#   oddities, this set is kept separate from the existing subscribers set on
# ==============================================================================

diff -r 920a06a52b8b -r fbdc7fd9cb96 docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Fri Jul 24 20:24:33 2020 +0000
+++ b/docshell/base/BrowsingContext.cpp	Fri Jul 24 20:24:36 2020 +0000
@@ -379,7 +379,7 @@
   uint64_t originId = 0;
   if (aOriginProcess) {
     originId = aOriginProcess->ChildID();
-    aGroup->EnsureSubscribed(aOriginProcess);
+    aGroup->EnsureHostProcess(aOriginProcess);
   }
 
   MOZ_LOG(GetLog(), LogLevel::Debug,